migration/model.go

```diff
@@ -0,0 +1,101 @@
+package migration
+
+import (
+	"encoding/binary"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+)
+
+var _ orm.CloneableData = (*Schema)(nil)
+
+func init() {
+	MustRegister(1, &Schema{}, NoModification)
+}
+
+func (s *Schema) Validate() error {
+	if s.Version < 1 {
+		return errors.Wrap(errors.ErrInvalidModel, "version must be greater than zero")
+	}
+	if s.Pkg == "" {
+		return errors.Wrap(errors.ErrInvalidModel, "pkg is required")
+	}
+	return nil
+}
+
+func (s *Schema) Copy() orm.CloneableData {
+	return &Schema{
+		Metadata: s.Metadata.Copy(),
+		Version:  s.Version,
+		Pkg:      s.Pkg,
+	}
+}
+
+// schemaID returns a detereministic ID of this schema instance.
+func schemaID(pkg string, version uint32) []byte {
+	raw := make([]byte, len(pkg)+4)
+	copy(raw, pkg)
+	binary.LittleEndian.PutUint32(raw[len(pkg):], version)
```

Comment:
Why LittleEndian?
We use BigEndian everywhere else for the reason that it provides lexographical ordering.

Eg. If I do a range scan for `cash` o `casi`, I will get `cash:0x1`, `cash:0x2`, `cash:0x3` in that order. If I use LittleEndian, I have no guarantees... `cash:0x0100`, `cash:0x0001`, `cash:0x0101`

Please use BigEndian for any integers that will be used in a key (and in general for consistency) - 6 times in our code vs. 0 for LittleEndian

Reply:
Protobuf varint is kind of like a compressed little-endian, but we don't use that for any keys

Reply:
Usually I follow [a rule](https://twitter.com/rob_pike/status/819768645182103553) when deciding this. It makes sense to change to BigEndian.

Reply:
Done in https://github.com/iov-one/weave/pull/550/commits/f5511e5e0eefb19afb4101f52bb2fed2784df3f5